inst/builtin/bs5/shiny/_rules.scss

```diff
@@ -84,41 +100,60 @@ $bslib-checkbox-radio-margin-right: 0.35em !default;
       font-size: 0.9rem;
       line-height: 1.375rem;
     }
+    .bslib-sidebar-layout > .sidebar {
+      border-top: var(--bslib-sidebar-border); 
+    }
   }
+
 }
 
 // TODO: move this logic into a "core" bundle that comes with any theme
 // (but you only get the shadows by default in the preset theme)
 @if $bslib-enable-shadows {
 
   :root {
-    --bslib-shiny-preset-main-bg: rgba(var(--bs-emphasis-color-rgb, 0,0,0), 0.05);
+    --bslib-shiny-preset-main-bg: rgba(var(--#{$prefix}emphasis-color-rgb, 0,0,0), 0.03);
   }
 
-  .bslib-page-sidebar {
-    --bslib-page-sidebar-title-color: var(--bs-body-color);
-    --bslib-page-sidebar-title-bg: var(--bs-body-bg);
+  .bslib-page-sidebar, .bslib-page-navbar {
+    .bslib-sidebar-layout {
+      --bslib-sidebar-bg: var(--#{$prefix}body-bg);
+      --bslib-sidebar-main-bg: var(--#{$prefix}body-bg);
+    }
+  }
 
-    > .bslib-sidebar-layout:first-of-type > .main {
+  .bslib-page-sidebar {
+    --bslib-page-sidebar-title-bg: transparent;
+    --bslib-page-sidebar-title-color: var(--#{$prefix}body-color);
+    
+    > .bslib-sidebar-layout > .main {
       background-color: var(--bslib-shiny-preset-main-bg);
       @include bslib-box-shadows;
     }
   }
 
   .bslib-page-navbar {
-    background-color: var(--bslib-shiny-preset-main-bg);
 
-    // 1st selector is for page_navbar(sidebar = ...)
-    // 2nd selector is for page_navbar(nav_panel("A", layout_sidebar()))
-    > *, .tab-pane {
-      > .bslib-sidebar-layout:first-of-type > .main {
-        @include bslib-box-shadows;
-      }
+    > .navbar {
+      --bslib-navbar-default-bg: transparent;
+      --bslib-navbar-inverse-bg: var(--#{$prefix}body-color);
     }
-  }
 
-  .bslib-sidebar-layout.bslib-sidebar-layout {
-    --bslib-sidebar-bg: var(--bs-body-bg);
+    > div {
+      // Since we're using a transparent navbar, we need to (generally) add a border-top
+      border-top: $card-border-width solid $card-border-color;
```

Comment:
This rule feels a little weird to me. Like it should really be a property of the navbar? I can see why its written this way since it needs to depend on the presence of the sidebar, it just tripped me up a bit.

Reply:
Yea, I definitely see your point. The main reason why I did it that way was to keep that logic close to the `border-top: none` below it, which is there to avoid a double-border situation [when these cases apply](https://github.com/rstudio/bslib/blob/f4a36e845bb78a326fcbe63994ae630fb2c5c35e/inst/components/scss/page_navbar.scss#L20-L39)